dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLMemoryStore.cs

```diff
@@ -0,0 +1,438 @@
+    private readonly string _databaseName;
+
+    /// <summary>
+    /// Initiates a AzureCosmosDBNoSQLMemoryStore instance using a Azure Cosmos DB connection string
+    /// and other properties required for vector search.
+    /// </summary>
+    /// <param name="connectionString">Connection string required to connect to Azure Cosmos DB.</param>
+    /// <param name="databaseName">The database name to connect to.</param>
+    /// <param name="vectorEmbeddingPolicy">The <see cref="VectorEmbeddingPolicy" /> to use if a collection is created. NOTE that embeddings will be stored in a property named 'embedding'.</param>
+    /// <param name="indexingPolicy">The <see cref="IndexingPolicy"/> to use if a collection is created. NOTE that embeddings will be stored in a property named 'embedding'.</param>
+    /// <param name="applicationName">The application name to use in requests.</param>
+    public AzureCosmosDBNoSQLMemoryStore(
+        string connectionString,
+        string databaseName,
+        VectorEmbeddingPolicy vectorEmbeddingPolicy,
+        IndexingPolicy indexingPolicy,
+        string? applicationName = null)
+    {
+        var options = new CosmosClientOptions
+        {
+            ApplicationName = applicationName ?? HttpHeaderConstant.Values.UserAgent,
+            Serializer = new CosmosSystemTextJsonSerializer(JsonSerializerOptions.Default),
+        };
+        this._cosmosClient = new CosmosClient(connectionString, options);
+        this._databaseName = databaseName;
+        this._vectorEmbeddingPolicy = vectorEmbeddingPolicy;
+        this._indexingPolicy = indexingPolicy;
+    }
+
+    /// <summary>
+    /// Initiates a AzureCosmosDBNoSQLMemoryStore instance using a <see cref="CosmosClient"/> instance
+    /// and other properties required for vector search.
+    /// </summary>
+    /// <param name="cosmosClient">An existing <see cref="CosmosClient"/> to use. NOTE: This must support serializing with
+    /// System.Text.Json, not the default Cosmos serializer.</param>
+    /// <param name="databaseName">The database name to operate against.</param>
+    /// <param name="vectorEmbeddingPolicy">The <see cref="VectorEmbeddingPolicy" /> to use if a collection is created. NOTE that embeddings will be stored in a property named 'embedding'.</param>
+    /// <param name="indexingPolicy">The <see cref="IndexingPolicy"/> to use if a collection is created. NOTE that embeddings will be stored in a property named 'embedding'.</param>
+    /// <param name="applicationName">The application name to use in requests.</param>
+    public AzureCosmosDBNoSQLMemoryStore(
+        CosmosClient cosmosClient,
+        string databaseName,
+        VectorEmbeddingPolicy vectorEmbeddingPolicy,
+        IndexingPolicy indexingPolicy,
+        string? applicationName = null)
+    {
+        this._cosmosClient = cosmosClient;
+        this._cosmosClient.ClientOptions.ApplicationName = applicationName;
```

Comment:
Cosmos DB folks - will this work after the `CosmosClient` has been created?

Reply:
Answering my own question - this will work unless the `ComosClient` already has a connection established, after which it would be ignored. I think that's okay for this case.